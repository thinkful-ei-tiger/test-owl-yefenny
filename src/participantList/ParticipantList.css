.participant-list ul li {
    list-style: none;
    margin: 15px;
}