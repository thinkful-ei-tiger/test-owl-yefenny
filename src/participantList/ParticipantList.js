import React, { Component } from 'react';
import './ParticipantList.css';
import Participant from '../participant/Participant';

export default class ParticipantList extends Component {
  render() {
    const participants = this.props.participants
      .sort((a, b) => b.inSession - a.inSession)
      .map((participant) => {
        return (
          <li key={participant.id}>
            <Participant
              avatar={participant.avatar}
              name={participant.name}
              inSession={participant.inSession}
              onStage={participant.onStage}
            />
          </li>
        );
      });
    return (
      <div className='participant-list'>
        <ul>{participants}</ul>
      </div>
    );
  }
}
