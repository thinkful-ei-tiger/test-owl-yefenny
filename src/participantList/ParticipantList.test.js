import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ParticipantList from './ParticipantList';
import PARTICIPANTS from '../dummy';
describe('ParticipantList component', () => {
  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ParticipantList participants={PARTICIPANTS} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('Renders UI without problem ', () => {
    const tree = renderer
      .create(<ParticipantList participants={PARTICIPANTS} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
