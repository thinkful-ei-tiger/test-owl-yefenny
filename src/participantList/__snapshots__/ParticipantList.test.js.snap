// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantList component Renders UI without problem  1`] = `
<div
  className="participant-list"
>
  <ul>
    <li>
      <div
        className="participant"
      >
        <div
          className="participant-img"
        >
          <img
            alt="avatar"
            src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
          />
        </div>
        <div
          className="participant-details"
        >
          <h2>
            Koren Templeton
          </h2>
          <div
            className="participant-status"
          >
            <span
              className="circle bg-green"
            />
            <span
              className="status"
            >
              on stage
            </span>
          </div>
        </div>
      </div>
    </li>
    <li>
      <div
        className="participant"
      >
        <div
          className="participant-img"
        >
          <img
            alt="avatar"
            src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
          />
        </div>
        <div
          className="participant-details"
        >
          <h2>
            Caty Flucker
          </h2>
          <div
            className="participant-status"
          >
            <span
              className="circle bg-green"
            />
            <span
              className="status"
            >
              in session
            </span>
          </div>
        </div>
      </div>
    </li>
    <li>
      <div
        className="participant"
      >
        <div
          className="participant-img"
        >
          <img
            alt="avatar"
            src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
          />
        </div>
        <div
          className="participant-details"
        >
          <h2>
            Frank Runciman
          </h2>
          <div
            className="participant-status"
          >
            <span
              className="circle bg-green"
            />
            <span
              className="status"
            >
              in session
            </span>
          </div>
        </div>
      </div>
    </li>
    <li>
      <div
        className="participant"
      >
        <div
          className="participant-img"
        >
          <img
            alt="avatar"
            src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
          />
        </div>
        <div
          className="participant-details"
        >
          <h2>
            Ashla Attwool
          </h2>
          <div
            className="participant-status"
          >
            <span
              className="circle bg-green"
            />
            <span
              className="status"
            >
              on stage
            </span>
          </div>
        </div>
      </div>
    </li>
    <li>
      <div
        className="participant"
      >
        <div
          className="participant-img"
        >
          <img
            alt="avatar"
            src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
          />
        </div>
        <div
          className="participant-details"
        >
          <h2>
            Axe Kubicka
          </h2>
          <div
            className="participant-status"
          >
            <span
              className="circle bg-grey"
            />
            <span
              className="status"
            >
              left session
            </span>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
